source/tutorials/sink-connector.txt

```diff
@@ -0,0 +1,194 @@
+
+      Paste the following configuration information into the file and save
+      your changes:
+
+      .. code-block::
+         :copyable: true
+         :emphasize-lines: 5-7
+
+         { "name": "mongo-sink", "config": {
+           "connector.class": "com.mongodb.kafka.connect.MongoSinkConnector",
+           "topics": "Tutorial2.pets",
+           "connection.uri": "mongodb://mongo1",
+           "key.converter": "org.apache.kafka.connect.storage.StringConverter",
+           "value.converter": "org.apache.kafka.connect.json.JsonConverter",
+           "value.converter.schemas.enable": false,
+           "database": "Tutorial2",
+           "collection": "pets"
+           }
+         }
+
+      .. note::
+
+         The highlighted lines in the configuration properties specify
+         **converters** which instruct the connector how to translate the data
+         from Kafka.
+
+      Run the following command in the shell to start the sink connector
+      using the configuration file you created:
+
+      .. code-block::
+         :copyable: true
+
+         cx simplesink.json
+
+      .. note::
+
+         The ``cx`` command is a custom script included in the tutorial
+         development environment. This script runs the following
+         equivalent request to the Kafka Connect REST API to create a new
+         connector:
+
+         .. code-block:: bash
+            :copyable: true
+
+            curl -X POST -H "Content-Type: application/json" -d @simplesink.json http://connect:8083/connectors -w "\n"
+
+      Run the following command in the shell to check the status of the
+      connectors:
+
+      .. code-block:: bash
+         :copyable: true
+
+         status
+
+      If your sink connector started successfully, you should see the
+      following output:
+
+      .. code-block:: none
+         :copyable: false
+
+         Kafka topics:
+         ...
+         The status of the connectors:
+
+         sink  |  mongo-sink  |  RUNNING  |  RUNNING  |  com.mongodb.kafka.connect.MongoSinkConnector
+
+         Currently configured connectors
+
+         [
+           "mongo-sink"
+         ]
+         ...
+
+   .. step:: Write Data to a Kafka Topic
+
+      In the same shell, create a Python script to open a change stream using
+      the PyMongo driver.
+
+      .. code-block:: bash
+         :copyable: true
+
+         nano kafkawrite.py
+
+      Paste the following code into the file and save your changes:
+
+      .. code-block::
+         :copyable: true
+
+         from kafka import KafkaProducer
+         import json
+         from json import dumps
+
+         p = KafkaProducer(bootstrap_servers=['broker:29092'],value_serializer=lambda x:dumps(x).encode('utf-8'))
+
+         data={'name':'roscoe'}
+
+         p.send('Tutorial2.pets', value=data)
+
+         p.flush()
+
+      Run the Python script:
+
+      .. code-block:: bash
+         :copyable: true
+
+         python3 kafkawrite.py
```

Comment:
issue: Should this stay a python script or be converted to an executable shell script with a shebang to keep the environment the "same" for the user following the tutorial?

Reply:
This runs on a bash shell of the Docker Container, so their environment should be the same for all users.

Reply:
Fair. I was thinking it might be more familiar to keep everything as a shell command without showing there's any python usage in there, but it's supposition.